Library/Homebrew/cmd/desc.rb

```diff
@@ -0,0 +1,59 @@
+require 'descriptions'
+require 'cmd/search'
+
+USAGE =<<EOM
+Print formula descriptions.
+
+Usage:
+
+    brew desc --decache             Delete the cache, if present
+    brew desc --recache             [Re]generate the cache
+    brew desc name [name ...]       Describe the named formula(e)
+    brew desc (-s|-n|-d) <string>   Search specifed fields for string
+    brew desc (-s|-n|-d) /<regex>/  Search specifed fields for regex
+EOM
+
+
+module Homebrew
+  def desc
+    options = ARGV.options_only
+
+    if options.include?("--decache")
+      Descriptions.delete_cache
+      options.delete("--decache")
+      cache_op = true
+    end
+
+    if options.include?("--recache")
+      Descriptions.generate_cache
+      options.delete("--recache")
+      cache_op = true
+    end
+
+    if options.empty?
+      if ARGV.named.empty?
+        print USAGE unless cache_op
```

Comment:
Isn’t `cache_op` always `false` here? The only way for it to be `true` would be that `options` include `--decache` or `--recache`, but to enter in this block it must be empty.